src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeNetworkFeeCryptoBaseUnit.tsx

```diff
@@ -0,0 +1,184 @@
+import { bchAssetId, CHAIN_NAMESPACE, fromChainId } from '@shapeshiftoss/caip'
+import { supportsETH } from '@shapeshiftoss/hdwallet-core'
+import type { SupportedTradeQuoteStepIndex } from '@shapeshiftoss/swapper'
+import {
+  getHopByIndex,
+  isExecutableTradeQuote,
+  SwapperName,
+  swappers,
+} from '@shapeshiftoss/swapper'
+import { skipToken, useQuery } from '@tanstack/react-query'
+import { getConfig } from 'config'
+import { useMemo } from 'react'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { assertUnreachable } from 'lib/utils'
+import { assertGetCosmosSdkChainAdapter } from 'lib/utils/cosmosSdk'
+import { assertGetEvmChainAdapter } from 'lib/utils/evm'
+import { assertGetSolanaChainAdapter } from 'lib/utils/solana'
+import { assertGetUtxoChainAdapter } from 'lib/utils/utxo'
+import { selectPortfolioAccountMetadataByAccountId } from 'state/slices/selectors'
+import {
+  selectActiveQuote,
+  selectActiveSwapperName,
+  selectConfirmedTradeExecution,
+  selectHopSellAccountId,
+  selectTradeSlippagePercentageDecimal,
+} from 'state/slices/tradeQuoteSlice/selectors'
+import { useAppSelector } from 'state/store'
+
+export const useTradeNetworkFeeCryptoBaseUnit = ({
+  hopIndex,
+  enabled = true,
+}: {
+  hopIndex: SupportedTradeQuoteStepIndex
+  enabled?: boolean
+}) => {
+  const wallet = useWallet().state.wallet
+  const slippageTolerancePercentageDecimal = useAppSelector(selectTradeSlippagePercentageDecimal)
+
+  const hopSellAccountIdFilter = useMemo(() => {
+    return {
+      hopIndex,
+    }
+  }, [hopIndex])
+
+  const confirmedTradeExecution = useAppSelector(selectConfirmedTradeExecution)
+  const sellAssetAccountId = useAppSelector(state =>
+    selectHopSellAccountId(state, hopSellAccountIdFilter),
+  )
+
+  const accountMetadataFilter = useMemo(
+    () => ({ accountId: sellAssetAccountId }),
+    [sellAssetAccountId],
+  )
+  const accountMetadata = useAppSelector(state =>
+    selectPortfolioAccountMetadataByAccountId(state, accountMetadataFilter),
+  )
+  const swapperName = useAppSelector(selectActiveSwapperName)
+  const tradeQuote = useAppSelector(selectActiveQuote)
+
+  const hop = useMemo(() => getHopByIndex(tradeQuote, hopIndex), [tradeQuote, hopIndex])
+  const swapper = useMemo(() => (swapperName ? swappers[swapperName] : undefined), [swapperName])
+
+  const quoteNetworkFeesCryptoBaseUnitQuery = useQuery({
+    queryKey: ['quoteNetworkFeesCryptoBaseUnit', tradeQuote],
+    refetchInterval: 15_000,
+    queryFn:
+      enabled &&
+      wallet &&
+      accountMetadata &&
+      tradeQuote &&
+      swapperName &&
+      sellAssetAccountId &&
+      swapper &&
+      hop &&
+      isExecutableTradeQuote(tradeQuote)
```

Comment:
q: does it mean that we fetch fees even if we just fetched them with the quote? couldn't we initially use the quote fees as cache data and refetch only 15 sec later? Would it be overengineered, any risk we notice?